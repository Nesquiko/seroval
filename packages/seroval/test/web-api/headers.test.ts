import { describe, it, expect } from 'vitest';
import 'node-fetch-native/polyfill';
import {
  crossSerialize,
  crossSerializeAsync,
  crossSerializeStream,
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../../src';

describe('Headers', () => {
  describe('serialize', () => {
    it('supports Headers', () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<Headers>(result);
      expect(back).toBeInstanceOf(Headers);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('serializeAsync', () => {
    it('supports Headers', async () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<Headers>>(result);
      expect(back).toBeInstanceOf(Headers);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSON', () => {
    it('supports Headers', () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<Headers>(result);
      expect(back).toBeInstanceOf(Headers);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSONAsync', () => {
    it('supports Headers', async () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<Headers>>(result);
      expect(back).toBeInstanceOf(Headers);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('crossSerialize', () => {
    it('supports Headers', () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = crossSerialize(example);
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Headers', () => {
        const example = new Headers([
          ['Content-Type', 'text/plain'],
          ['Content-Encoding', 'gzip'],
        ]);
        const result = crossSerialize(example, { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports Headers', async () => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      const result = await crossSerializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Headers', async () => {
        const example = new Headers([
          ['Content-Type', 'text/plain'],
          ['Content-Encoding', 'gzip'],
        ]);
        const result = await crossSerializeAsync(Promise.resolve(example), { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeStream', () => {
    it('supports Headers', async () => new Promise<void>((done) => {
      const example = new Headers([
        ['Content-Type', 'text/plain'],
        ['Content-Encoding', 'gzip'],
      ]);
      crossSerializeStream(Promise.resolve(example), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    describe('scoped', () => {
      it('supports Headers', async () => new Promise<void>((done) => {
        const example = new Headers([
          ['Content-Type', 'text/plain'],
          ['Content-Encoding', 'gzip'],
        ]);
        crossSerializeStream(Promise.resolve(example), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
    });
  });
});
