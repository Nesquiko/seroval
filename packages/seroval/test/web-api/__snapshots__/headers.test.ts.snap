// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Headers > crossSerialize > scoped > supports Headers 1`] = `"((self.$R=self.$R||{})[\\"example\\"]=[],($R=>$R[0]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"}))($R[\\"example\\"]))"`;

exports[`Headers > crossSerialize > supports Headers 1`] = `"(self.$R=self.$R||[],$R[0]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"}))"`;

exports[`Headers > crossSerializeAsync > scoped > supports Headers 1`] = `"((self.$R=self.$R||{})[\\"example\\"]=[],($R=>$R[0]=Promise.resolve($R[1]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"})))($R[\\"example\\"]))"`;

exports[`Headers > crossSerializeAsync > supports Headers 1`] = `"(self.$R=self.$R||[],$R[0]=Promise.resolve($R[1]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"})))"`;

exports[`Headers > crossSerializeStream > scoped > supports Headers 1`] = `"((self.$R=self.$R||{})[\\"example\\"]=[],($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R[\\"example\\"]))"`;

exports[`Headers > crossSerializeStream > scoped > supports Headers 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status=\\"success\\",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"})))($R[\\"example\\"])"`;

exports[`Headers > crossSerializeStream > supports Headers 1`] = `"(self.$R=self.$R||[],$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())"`;

exports[`Headers > crossSerializeStream > supports Headers 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status=\\"success\\",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"}))"`;

exports[`Headers > serialize > supports Headers 1`] = `"new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"})"`;

exports[`Headers > serializeAsync > supports Headers 1`] = `"Promise.resolve(new Headers({\\"content-encoding\\":\\"gzip\\",\\"content-type\\":\\"text/plain\\"}))"`;

exports[`Headers > toCrossJSON > supports Headers 1`] = `"{\\"t\\":25,\\"i\\":0,\\"e\\":{\\"k\\":[\\"content-encoding\\",\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"gzip\\"},{\\"t\\":1,\\"s\\":\\"text/plain\\"}],\\"s\\":2}}"`;

exports[`Headers > toCrossJSONAsync > supports Headers 1`] = `"{\\"t\\":12,\\"i\\":0,\\"s\\":1,\\"f\\":{\\"t\\":25,\\"i\\":1,\\"e\\":{\\"k\\":[\\"content-encoding\\",\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"gzip\\"},{\\"t\\":1,\\"s\\":\\"text/plain\\"}],\\"s\\":2}}}"`;

exports[`Headers > toCrossJSONStream > supports Headers 1`] = `"{\\"t\\":28,\\"i\\":0,\\"f\\":{\\"t\\":41,\\"i\\":1,\\"s\\":2}}"`;

exports[`Headers > toCrossJSONStream > supports Headers 2`] = `"{\\"t\\":29,\\"i\\":0,\\"a\\":[{\\"t\\":41,\\"i\\":3,\\"s\\":3},{\\"t\\":25,\\"i\\":2,\\"e\\":{\\"k\\":[\\"content-encoding\\",\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"gzip\\"},{\\"t\\":1,\\"s\\":\\"text/plain\\"}],\\"s\\":2}}]}"`;

exports[`Headers > toJSON > supports Headers 1`] = `"{\\"t\\":{\\"t\\":25,\\"i\\":0,\\"e\\":{\\"k\\":[\\"content-encoding\\",\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"gzip\\"},{\\"t\\":1,\\"s\\":\\"text/plain\\"}],\\"s\\":2}},\\"f\\":16383,\\"m\\":[]}"`;

exports[`Headers > toJSONAsync > supports Headers 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"s\\":1,\\"f\\":{\\"t\\":25,\\"i\\":1,\\"e\\":{\\"k\\":[\\"content-encoding\\",\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"gzip\\"},{\\"t\\":1,\\"s\\":\\"text/plain\\"}],\\"s\\":2}}},\\"f\\":16383,\\"m\\":[]}"`;
